carto/sql.py

```diff
@@ -0,0 +1,224 @@
+"""
+Module for the SQL API
+
+.. module:: carto.sql
+   :platform: Unix, Windows
+   :synopsis: Module for the SQL API
+
+.. moduleauthor:: Daniel Carrion <daniel@carto.com>
+.. moduleauthor:: Alberto Romeu <alrocar@carto.com>
+
+
+"""
+
+from .exceptions import CartoException
+
+SQL_API_URL = 'api/{api_version}/sql'
+SQL_BATCH_API_URL = 'api/{api_version}/sql/job/'
+
+MAX_GET_QUERY_LEN = 1024
+
+
+class SQLClient(object):
+    """
+    Allows you to send requests to CARTO's SQL API
+    """
+    def __init__(self, auth_client, api_version='v2'):
+        """
+        :param auth_client: Auth client to make authorized requests, such as
+                            APIKeyAuthClient
+        :param api_version: Current version is 'v2'. 'v1' can be used to avoid
+                            caching, but it's not guaranteed to work
+        :type auth_client: :class:`carto.auth.APIKeyAuthClient`
+        :type api_version: str
+
+        :return:
+        """
+        self.auth_client = auth_client
+        self.api_url = SQL_API_URL.format(api_version=api_version)
+
+        self.api_key = self.auth_client.api_key \
+            if hasattr(self.auth_client, "api_key") else None
+        self.base_url = self.auth_client.base_url
+        self.username = self.auth_client.username \
```

Comment:
this (and the ones around) can be replaced by `getattr(self.auth_client, 'username', None)`